### PR TITLE
Add how-to clone the superproject to README.md

### PR DESCRIPTION
**Pull request by mloskot**
Simplest way to address these issues https://lists.boost.org/Archives/boost/2020/04/248731.php

If one aims to clone a repo from GitHub, then README.md is the obvious muscle-memorised place to look at.

-----

By the way, a curated `CONTRIBUTING.md` could replace the mess on the Wiki as a single source of truth.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,3 +9,38 @@ and does not require attribution for binary use.
...
+or checkout the `develop` branch with the latest development snapshot.
+
+```console
+git clone --branch develop --recurse-submodules https://github.com/boostorg/boost.git boost-root
+```
+
+Alternatively, clone just the superproject repository, then clone a desired submodule together with its dependencies:
+
+```console
+git clone --recurse-submodules https://github.com/boostorg/boost.git boost-root
```
Comment (Flamefire) [inline README.md:30]:
```suggestion
git clone https://github.com/boostorg/boost.git boost-root
```

File: README.md [hunk 2]
```diff
@@ -9,3 +9,38 @@ and does not require attribution for binary use.
...
+cd boost-root
+git submodule update --init tools/boostdep
+python tools/boostdep/depinst/depinst.py filesystem
+```
+
+Alternatively, clone just the superproject repository, then manually pick individual submodules:
+
+```console
+git clone --recurse-submodules https://github.com/boostorg/boost.git boost-root
```
Comment (Flamefire) [inline README.md:39]:
```suggestion
git clone https://github.com/boostorg/boost.git boost-root
```

File: README.md (hunk 1 above)
Comment (mloskot) [reply README.md]: Ha, amazing! Thanks very much

File: README.md (hunk 2 above)
Comment (mloskot) [reply README.md]: Fixed too

File: README.md [hunk 3]
```diff
@@ -9,3 +9,38 @@ and does not require attribution for binary use.
...
+```console
+git clone --branch develop --recurse-submodules https://github.com/boostorg/boost.git boost-root
+```
+
+Alternatively, clone just the superproject repository, then clone a desired submodule together with its dependencies:
+
+```console
+git clone https://github.com/boostorg/boost.git boost-root
+cd boost-root
+git submodule update --init tools/boostdep
```
Comment (Kojoley) [inline README.md:32]: one-liner `git clone --recurse-submodules="tools/boostdep" https://github.com/boostorg/boost.git boost-root`

**Comment by glenfe** [comment]
Maybe we can link from the README.md to a page on boostorg/wiki and you can edit that page more regularly.

File: README.md (hunk 3 above)
Comment (mloskot) [reply README.md:32]: The aim was not to show off a clever git fu but present self-explanatory examples that even those who just know git basics can understand yhe differences between 2-3 possible or common workflows.

**Comment by mloskot** [comment]
I'd be against wiki. Wikis either rot or drift away. PR driven updates ensure a second pair of eyes do review before merge.

**Pull request merged**